### PR TITLE
Update Features to declare settings as JSON

### PR DESCRIPTION
**Pull request by felipeelia**
### Description of the Change

Closes #

### Changelog Entry

> Added - New feature
> Changed - Existing functionality
> Deprecated - Soon-to-be removed feature
> Removed - Feature
> Fixed - Bug fix
> Security - Vulnerability

### Credits

Props @username, @username2, ...

### Checklist:

- [ ] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: includes/classes/Feature/DidYouMean/DidYouMean.php [hunk 1]
```diff
@@ -415,4 +417,35 @@ public function the_output( $query = null ) {
...
+		$this->settings_schema = [
+			[
+				'default' => false,
+				'key'     => 'search_behavior',
+				'label'   => __( 'Search behavior when no result is found', 'elasticpress' ),
+				'options' => [
+					[
+						[
+							'label' => __( 'Display the top suggestion', 'elasticpress' ),
+							'key'   => 0,
```
Comment (JakePT) [inline includes/classes/Feature/DidYouMean/DidYouMean.php]: @felipeelia I think this should be `value`? That's what's used elsewhere and it makes it easier to pass into the various core components that accept options.

File: includes/classes/Feature.php [hunk 2]
```diff
@@ -503,4 +511,60 @@ public function is_available() : bool {
...
+	public function get_json() {
+		$feature_desc = [
+			'slug'            => $this->slug,
+			'title'           => $this->get_title(),
+			'shortTitle'      => $this->get_short_title(),
+			'summary'         => $this->summary,
+			'docsUrl'         => $this->docs_url,
+			'defaultSettings' => $this->default_settings,
+			'order'           => $this->order,
+			'isAvailable'     => $this->is_available(),
```
Comment (JakePT) [inline includes/classes/Feature.php]: @felipeelia Just want to double check that a boolean `isAvailable` is sufficient for what we need on the front-end, or whether we need the full requirements status. I believe there's currently a different status color and message based on the requirements code, which wouldn't be possible with just this boolean.

File: includes/classes/Feature/Autosuggest/Autosuggest.php [hunk 3]
```diff
@@ -855,6 +857,76 @@ public function intercept_remote_request() {
...
+
+		if ( Utils\is_epio() ) {
+			$this->maybe_add_epio_settings_schema();
+			return;
+		}
+
+		$set_in_wp_config = defined( 'EP_AUTOSUGGEST_ENDPOINT' ) && EP_AUTOSUGGEST_ENDPOINT;
+
+		$this->settings_schema[] = [
+			'editable' => ! $set_in_wp_config,
```
Comment (JakePT) [inline includes/classes/Feature/Autosuggest/Autosuggest.php]: @felipeelia Can we use `readonly` for this? It maps a little better to how the field will be treated on the front end.

**Review by JakePT — CHANGES_REQUESTED** [review]
Added a few comments.

File: includes/classes/Feature/DidYouMean/DidYouMean.php (hunk 1 above)
Comment (felipeelia) [reply includes/classes/Feature/DidYouMean/DidYouMean.php]: Thanks @JakePT, fixed in [c105e34](https://github.com/10up/ElasticPress/pull/3655/commits/c105e34d5787be58532feb7c132ca9c75fc00231).

File: includes/classes/Feature/Autosuggest/Autosuggest.php (hunk 3 above)
Comment (felipeelia) [reply includes/classes/Feature/Autosuggest/Autosuggest.php]: Thanks @JakePT, adjusted in [c105e34](https://github.com/10up/ElasticPress/pull/3655/commits/c105e34d5787be58532feb7c132ca9c75fc00231).

File: includes/classes/Feature.php (hunk 2 above)
Comment (felipeelia) [reply includes/classes/Feature.php]: I kept that (imagining it'll make your life a bit easier) but I'm now also including the code and the message(s) from the requirement_status method.

**Review by JakePT — APPROVED** [review]

**Pull request merged**